src/Subscriptions/SubscriptionRouter.php

```diff
@@ -14,14 +14,16 @@ class SubscriptionRouter
      */
     public function pusher($router): void
     {
+        $middleware = config('lighthouse.subscriptions.route.middleware');
+
         $router->post('graphql/subscriptions/auth', [
             'as' => 'lighthouse.subscriptions.auth',
             'uses' => SubscriptionController::class.'@authorize',
-        ]);
+        ])->middleware($middleware);
```

Comment:
I think it is misleading to add an array of route options and only apply one key of it.

Can we just make this a route group, same as in the base config?

Reply:
@spawnia The middleware method accepts an array as the first parameter. So this will work as it does on the `/graphql/ route middleware.

Reply:
I belive what @spawnia is suggesting is having it share the same behaviour as the config of the regular routes. That way you will be able to apply more than just middleware to the subscription routes. See https://github.com/nuwave/lighthouse/blob/master/src/Support/Http/routes.php

Reply:
Ah, I miss understood the request at first. I've updated it, and also changed it to use the `route_name` config for the base part of the path.